sdk/core/azure-core-management/src/test/java/com/azure/core/management/implementation/polling/LROPollerTests.java

```diff
@@ -338,36 +416,51 @@ public String getName() {
         try {
             final ProvisioningStateLroServiceClient client = RestProxy.create(ProvisioningStateLroServiceClient.class,
                 createHttpPipeline(lroServer.port()), SERIALIZER);
-
-            PollerFlux<PollResult<Resource>, Resource> lroFlux
-                = PollerFactory.create(SERIALIZER, new HttpPipelineBuilder().build(), Resource.class, Resource.class,
-                    POLLING_DURATION, newLroInitFunction(client));
-
-            StepVerifier.create(lroFlux).expectSubscription().expectNextMatches(response -> {
-                PollResult<Resource> pollResult = response.getValue();
-                return response.getStatus() == LongRunningOperationStatus.SUCCESSFULLY_COMPLETED
-                    && pollResult != null
-                    && pollResult.getValue() != null
-                    && pollResult.getValue().id() != null;
-            }).verifyComplete();
-
-            AsyncPollResponse<PollResult<Resource>, Resource> asyncPollResponse = lroFlux.blockLast();
-            Assertions.assertNotNull(asyncPollResponse);
-
-            Resource result = asyncPollResponse.getFinalResult().block();
-            Assertions.assertNotNull(result);
-            Assertions.assertNotNull(result.id());
-            Assertions.assertEquals("v1weidxu", result.name());
-            Assertions.assertEquals("Microsoft.KeyVault/vaults", result.type());
+            if (syncStack) {
+                SyncPoller<PollResult<Resource>, Resource> lroPoller
+                    = SyncPollerFactory.create(SERIALIZER, new HttpPipelineBuilder().build(), Resource.class,
+                        Resource.class, POLLING_DURATION, () -> newLroInitResponse(client));
+
+                PollResponse<PollResult<Resource>> pollResponse = lroPoller.poll();
+                Assertions.assertEquals(LongRunningOperationStatus.SUCCESSFULLY_COMPLETED, pollResponse.getStatus());
```

Comment:
Are we able to verify that this won't need any delay on poll (one-time manual verify is fine) as this case is 200 directly, if we do e.g. `lroPoller.getFinalResult()`?

Reply:
It's quite tricky since first poll in `SyncPoller.getFinalResult` won't trigger any delay: https://github.com/Azure/azure-sdk-for-java/blob/5f693e54afe6ca90faea90361355c3d19b0f8861/sdk/core/azure-core/src/main/java/com/azure/core/util/polling/PollingUtil.java#L58

Added a verification that `PUT` url will only be called once(which is initial request). Any additional polling or getting result will cause exception.

Reply:
I mainly mean the delay (not the call). E.g. whether it do PUT, delay 30sec, then poll (no request, just check and find it success).

Reply:
added